app/src/main/java/com/github/geohunt/app/database/FirebaseDatabaseWrapper.java

```diff
@@ -0,0 +1,81 @@
+package com.github.geohunt.app.database;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+
+public class FirebaseDatabaseWrapper implements Database {

```

Comment:
I think that `FirebaseDatabaseAdapter` would be a more appropriate name, and similarly for `FirebaseDatabasePathAdapter` instead of `FirebaseDatabasePathWrapper`.